File: ios/craftd-swiftui/Sources/craftd-swiftui/presentation/builder/CraftDBuilder.swift
import SwiftUI
import Foundation

public protocol CraftDBuilder {
    var key: String { get }
    
    func craft(
        model: SimpleProperties,
        listener: @escaping CraftDViewListener
    ) -> any View
}

public extension CraftDBuilder {
    var key: String {
        String("\(Self.self)".replacingOccurrences(of: "Builder", with: ""))
    }
    
    var decoder: JSONDecoder {
        JSONDecoder()
    }
}

File: android_kmp/craftd-compose/src/main/java/com/github/codandotv/craftd/compose/builder/CraftDBuilder.kt
package com.github.codandotv.craftd.compose.builder

import androidx.compose.runtime.Composable
import com.github.codandotv.craftd.androidcore.data.model.base.SimpleProperties
import com.github.codandotv.craftd.androidcore.presentation.CraftDViewListener

interface CraftDBuilder {
    val key: String
    @Composable
    fun craft(model: SimpleProperties, listener: CraftDViewListener)
}